packages/backend/src/gameengine/gameobjects/abstract-game-object.ts

```diff
@@ -6,26 +6,37 @@ export abstract class AbstractGameObject {
   private static nextId = 1;
   private readonly _id = AbstractGameObject.nextId++;
 
-  constructor(startPosition: Position) {
-    this.position = startPosition;
+  constructor(startPosition: Position | null) {
+    if (startPosition) {
+      this.positions.push(startPosition);
+    }
   }
 
-  get x(): number {
-    return this.position.x;
+  get x(): number | null {
+    return this.position ? this.position.x : null;
   }
 
-  get y(): number {
-    return this.position.y;
+  get y(): number | null {
+    return this.position ? this.position.y : null;
   }
 
   get id(): number {
     return this._id;
   }
-  get position(): Position {
-    return { ...this.positions[this.positions.length - 1] };
+
+  get position(): Position | null {
+    if (this.positions.length > 0) {
+      return { ...this.positions[this.positions.length - 1] };
+    }
+    return null;
   }
-  set position(newPosition: Position) {
-    this.positions.push(newPosition);
+  set position(newPosition: Position | null) {
+    if (newPosition) {
+      this.positions.push(newPosition);
+    } else {
+      // Is this reasonable? Every getter needs a setter.
+      this.clearPositions();
+    }
```

Comment:
nu har det inte med den här pren att göra men varför sparar vi en lista av positioner? 🤔

Reply:
Det kan jag tyvärr inte svara på. Misstänker för att vi vill ha bottens historia?
Jag konstaterade bara att när vi hämtar nuvarande position så tar vi det sista i listan så jag tänkte att vi konsekvent borde arbeta mot listan.

Reply:
får kolla i koden sen, skit samma. ett sidetrack i pren, it looks good to me

Reply:
Jag tänkte kolla igenom PRn en extra gång sen för har lite tankar om beteendet men just nu är jag lite i första halvan av "make it work, make it right"-läge